modules/@apostrophecms/asset/lib/webpack/utils.js

```diff
@@ -276,7 +371,57 @@ async function verifyBundlesEntryPoints (bundles) {
   return packedFilesByBundle;
 };
 
-function formatConfigs (chain, webpackConfigs) {
+// Normalize config:
+// - from { moduleName: 'main' }
+//    to [{ main: true, name: 'moduleName', target: 'main' }]
+// - from { moduleName: bundleName }
+//    to [{ main: false, name: 'moduleName', target: bundleName }]
+// - from { 'moduleName:sourceBundle': 'main' }
+//    to [{ main: true, name: 'moduleName', source: 'sourceBundle', target: 'main' }]
+// - from { 'moduleName:sourceBundle': targetBundle }
+//    to [{ main: false, name: 'moduleName', source: 'sourceBundle', target: targetBundle }]
+function formatRebundleConfig(mappingConfig = {}) {
+  const result = Object.keys(mappingConfig).reduce((transformed, key) => {
+    const main = mappingConfig[key] === 'main';
+    const [ moduleName, sourceBundle ] = key.split(':');
+    return [
+      ...transformed,
+      {
+        name: moduleName,
+        source: sourceBundle,
+        target: mappingConfig[key],
+        main
+      }
+    ];
+  }, []);
+  // Better panic than sorry!
+  verifyRebundleConfig(result);
+  return result;
+}
+
+// Expects formatted by formatRebundleConfig() `asset.options.rebundleModules`
+function verifyRebundleConfig(config = []) {
+  const targeted = [];
+  const catchAllModules = [];
+  for (const entry of config) {
+    if (!entry.source) {
+      catchAllModules.push(entry.name);
+    } else {
+      targeted.push(entry);
+    }
+  }
+  for (const entry of targeted) {
+    if (catchAllModules.includes(entry.name)) {
+      throw new Error(
+        'Invalid apos.asset.options.rebundleModules: ' +
+        `module ${entry.name} can't have more re-bundle options when` +
```

Comment:
As a developer I wouldn't know what to change in my code after reading this. What's a catch-all directive? Can this error actually happen as far as you know or is this more of a "we have a bug, sorry" message?

Reply:
Fixed to: `Error: Invalid apos.asset.options.rebundleModules: "bundle-page: main" conflicts with "bundle-page:extra: new"`